wasmCloud/wasmCloud
# chore(Feature)::> Centralize directory handling with ProjectPaths configuration

## Feature or Problem
> Refactors directory handling to remove direct dependencies on std::env::current_dir() by introducing a centralized ProjectPaths configuration. This improves code reusability, testability, and makes directory management more explicit and maintainable across the codebase. Addresses issues with unergonomic helper code in wash-lib.


## Related Issues
closes #3800 


## Release Information
next


### Manual Verification
1. Created new ProjectPaths configuration and updated affected files
2. Ran existing test suite to ensure no regressions
3. Manually tested the following commands:
   - wash wit build
   - wash wit deps
   - wash dev
4. Verified custom directory paths work correctly
5. Checked that all path resolutions remain consistent with previous behavior
